packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -130,6 +131,13 @@ class Dart2JSTarget extends Target {
       ? PackageMap.globalGeneratedPackagesPath
       : PackageMap.globalPackagesPath;
     final File outputFile = environment.buildDir.childFile('main.dart.js');
+
+    // Dart defines are encoded as a JSON array, so we need to parse it. Also JSON
+    // parser returns List<dynamic>, so we need to cast it.
+    final List<String> dartDefines = environment.defines.containsKey(kDartDefines)
```

Comment:
catch the parse exception and rethrow with an error message

Reply:
A parser exception is not possible, is it?

Reply:
`flutter assemble -dDartDefines=[23,` would throw I believe

Reply:
Ah, gotcha. Fixing.

Reply:
Done.